pkg/api/customization/nodepool/validator.go

```diff
@@ -2,22 +2,51 @@ package nodepool
 
 import (
 	"fmt"
+	"strings"
 
 	"github.com/rancher/norman/api/access"
 	"github.com/rancher/norman/httperror"
 	"github.com/rancher/norman/types"
+	v3 "github.com/rancher/types/apis/management.cattle.io/v3"
 	mgmtSchema "github.com/rancher/types/apis/management.cattle.io/v3/schema"
 	mgmtclient "github.com/rancher/types/client/management/v3"
 )
 
-func Validator(request *types.APIContext, schema *types.Schema, data map[string]interface{}) error {
-	nodetemplateID, _ := data["nodeTemplateId"].(string)
-	if err := access.ByID(request, &mgmtSchema.Version, mgmtclient.NodeTemplateType, nodetemplateID, nil); err != nil {
+type Validator struct {
+	NodePoolLister v3.NodePoolLister
+}
+
+func (v *Validator) Validator(request *types.APIContext, schema *types.Schema, data map[string]interface{}) error {
+	// does we have access to the nodepool?
+	if err := access.ByID(request, &mgmtSchema.Version, mgmtclient.NodePoolType, request.ID, nil); err != nil {
```

Comment:
You shouldn't have to check if the user has access to the nodepool in the nodepool validator. I believe an auth check is already being done for every resource request. We usually do this when another resource is being referenced in the body of the request..